pkg/transforms/mqttsecret.go

```diff
@@ -0,0 +1,236 @@
+package transforms
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+	"github.com/edgexfoundry/app-functions-sdk-go/pkg/util"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+// MQTTSecretSender ...
+type MQTTSecretSender struct {
+	client         MQTT.Client
+	mqttConfig     MQTTSecretConfig
+	persistOnError bool
+	opts           *MQTT.ClientOptions
+	lastRetrieved  time.Time
+}
+
+// MQTTSecretConfig ...
+type MQTTSecretConfig struct {
+	// BrokerAddress should be set to the complete broker address i.e. mqtts://mosquitto:8883/mybroker
+	BrokerAddress string
+	// ClientId to connect with the broker with.
+	ClientId string
+	// The name of the path in vault to retrieve your secrets
+	SecretPath string
+	// AutoReconnect indicated whether or not to retry connection if disconnected
+	AutoReconnect bool
+	// Topic that you wish to publish to
+	Topic          string
+	QoS            byte
+	Retain         bool
+	SkipCertVerify bool
+	// AuthMode indicates what to use when connecting to the broker. Options are "none", "ca" , "usernamepassword", "cert".
```

Comment:
i was trying to have a clearer difference between `cert` and `ca` since `cacert` and `cert` seems like they could be easily mixed up. But I'll change it.

Reply:
How about also `cert` => `clientcert`, if that is the correct name.